src/features/defi/providers/idle/components/IdleManager/Deposit/IdleDeposit.tsx

```diff
@@ -0,0 +1,135 @@
+import { DepositContext } from './DepositContext'
+import { initialState, reducer } from './DepositReducer'
+
+export const IdleDeposit = () => {
+  const { idle: api } = useIdle()
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const translate = useTranslate()
+  const toast = useToast()
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const chainAdapterManager = getChainAdapterManager()
+  const { chainId, contractAddress: vaultAddress, assetReference } = query
+
+  const assetNamespace = 'erc20'
+  const assetId = toAssetId({ chainId, assetNamespace, assetReference })
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  // user info
+  const chainAdapter = chainAdapterManager.get(chainId)
+  const { state: walletState } = useWallet()
+  const loading = useSelector(selectPortfolioLoading)
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!(walletState.wallet && vaultAddress && chainAdapter && api)) return
+        const [address, opportunity] = await Promise.all([
+          chainAdapter.getAddress({ wallet: walletState.wallet }),
+          api.findByOpportunityId(
+            toAssetId({ chainId, assetNamespace, assetReference: vaultAddress }),
+          ),
+        ])
+        if (!opportunity) {
+          return toast({
+            position: 'top-right',
+            description: translate('common.somethingWentWrongBody'),
+            title: translate('common.somethingWentWrong'),
+            status: 'error',
+          })
+        }
+
+        dispatch({ type: IdleDepositActionType.SET_USER_ADDRESS, payload: address })
+        dispatch({ type: IdleDepositActionType.SET_OPPORTUNITY, payload: opportunity })
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('IdleDeposit error:', error)
+      }
+    })()
+  }, [api, chainAdapter, vaultAddress, walletState.wallet, translate, toast, chainId])
+
+  const handleBack = () => {
+    history.push({
+      pathname: location.pathname,
+      search: qs.stringify({
+        ...query,
+        modal: DefiAction.Overview,
+      }),
+    })
+  }
+
+  const StepConfig: DefiStepProps = useMemo(() => {
+    return {
+      [DefiStep.Info]: {
+        label: translate('defi.steps.deposit.info.title'),
+        description: translate('defi.steps.deposit.info.description', { asset: asset.symbol }),
+        component: Deposit,
+      },
+      [DefiStep.Approve]: {
+        label: translate('defi.steps.approve.title'),
+        component: Approve,
+      },
+      [DefiStep.Confirm]: {
+        label: translate('defi.steps.confirm.title'),
+        component: Confirm,
+      },
+      [DefiStep.Status]: {
+        label: 'Status',
+        component: Status,
+      },
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Same thing with these, copied from current implementation so let's keep it.
I'll remove them with a follow-up PR similar to https://github.com/shapeshift/web/pull/2324